benchs/Bench.hs

```diff
@@ -14,15 +15,69 @@ import           Crypto.Hash
 import qualified Crypto.KDF.PBKDF2 as PBKDF2
 import qualified Crypto.PubKey.ECC.Types as ECC
 import qualified Crypto.PubKey.ECC.Prim as ECC
+import           Crypto.Random
 
-import           Data.ByteArray (ByteArray)
+import           Data.ByteArray (ByteArray, Bytes)
 import qualified Data.ByteString as B
 
+import           System.IO.Unsafe (unsafePerformIO)
+
 import Number.F2m
 
+data HashAlg = forall alg . HashAlgorithm alg => HashAlg alg
+
 benchHash =
-    [ 
+    [ bgroup "1KB" $ map (doHashBench oneKB) hashAlgs
+    , bgroup "1MB" $ map (doHashBench oneMB) hashAlgs
     ]
+  where
+    doHashBench b (name, HashAlg alg) = bench name $ nf (hashWith alg) b
+
+    oneKB :: Bytes
+    oneKB = unsafePerformIO (getRandomBytes 1024)
```

Comment:
Yeah, that's me being extra cautious just in case there is a bias with some simple sequence. And also my lack of imagination about what to choose instead.

So I'll merge with random but anyone more convinced can change it.